api-scanner/src/test/java/net/corda/plugins/AnnotatedClassTest.java

```diff
@@ -13,14 +13,15 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 
 import static net.corda.plugins.CopyUtils.*;
 import static org.assertj.core.api.Assertions.*;
 import static org.junit.Assert.*;
 
 public class AnnotatedClassTest {
     @Rule
-    public final TemporaryFolder testProjectDir = new TemporaryFolder();
+    public final TemporaryFolder testProjectDir = new TemporaryFolder(Paths.get("build").toFile());
```

Comment:
This _shouldn't_ be necessary, because the `build.gradle` file at the project root contains the following:
```gradle
allprojects {
    ...
    tasks.withType(Test) {
        // Prevent the project from creating temporary files outside of the build directory.
        systemProperty 'java.io.tmpdir', buildDir.absolutePath
        ...
    }
    ...
}
```
My understanding of JUnit's `TemporaryFolder()` rule is that it is equivalent to:
```java
File.createTempFile("junit", "", null)
```
which means that it will use `java.io.tmpdir` as the parent directory. So what error are you getting on Windows? Is the value of `buildDir.absolutePath` wrong somehow?

Reply:
On windows, it is creating a temporary directory in my user account local data, nowhere near where the build is running, so all the relative directories are wrong.

Reply:
The point is that this _cannot_ be a Windows problem because `java.io.tmpdir` works fine on Windows. It sounds more like this system property isn't being set - perhaps IntelliJ isn't parsing the `build.gradle` files correctly?

Reply:
Having said all this, have you tested (say) the `KotlinLambdasTest`? Because if IntelliJ _is_ the problem becsuse it's only partially handling the `build.gradle` files, then I suspect it's not handling this either:
```gradle
processTestResources {
    filesMatching('**/kotlin-*/build.gradle') {
        expand(['kotlin_version': kotlin_version])
    }
}
```
which means that the Kotlin tests will fail. And if `java.io.tmpdir` isn't being set then `test.gradle.user/home` isn't going to be set either. And that means that each unit test using Kotlin will download the Kotlin artifacts from the Internet - _every time_!

Basically, these tests were written to run in Gradle. As indeed I thought the entire `corda-gradle-plugins` project is.

Reply:
The tests fail in gradle as well as intellij for the same reasons on my machine. I'm just investigating the `java.io.tmpdir` variable now.